pkg/schedule/operator/operator_controller.go

```diff
@@ -455,6 +458,27 @@ func (oc *Controller) checkAddOperator(isPromoting bool, ops ...*Operator) (bool
 	return reason != Expired, reason
 }
 
+// checkOperatorLightly checks whether the ops can be dispatched in Controller::pollNeedDispatchRegion.
+// The operators can't be dispatched for some scenarios, such as region disappeared, region changed ...
+// `region` is the target region of `op`.
+func (oc *Controller) checkOperatorLightly(op *Operator) (*core.RegionInfo, CancelReasonType) {
+	region := oc.cluster.GetRegion(op.RegionID())
+	if region == nil {
+		operatorCounter.WithLabelValues(op.Desc(), "not-found").Inc()
+		return nil, RegionNotFound
+	}
+
+	// It may be suitable for all kinds of operator but not merge-region.
+	// But to be cautions, it only takes effect on merge-region currently.
+	// If the version of epoch is changed, the region has been splitted or merged, and the key range has been changed.
+	// The changing for conf_version of epoch doesn't modify the region key range, skipt it.
+	if (op.Kind()&OpMerge != 0) && region.GetRegionEpoch().GetVersion() != op.RegionEpoch().GetVersion() {
```

Comment:
Do we need to compare the epoch like checkStaleOperator?

Reply:
`checkStaleOperator` only check the ConfVersion of region epoch which  maybe not resolve the issue https://github.com/tikv/pd/issues/7992. For example, "operator add split-region 258 --policy="approximate" will change the version of region 258, but not for ConfVersion.
Another reason is I don't know why limits the checkStaleOperator only to `DispatchFromHeartBeat` source https://github.com/tikv/pd/blob/master/pkg/schedule/operator/operator_controller.go#L120

Reply:
Yes, what I mean here is that is it enough to only check the version? Or we also need to check both the version and conf version?

Reply:
At begining, I checked both the version and conf-version. But I think the conf-version changed doesn't measn the key-range changed, the merge operator can still be executed sucessfully. At last I check only the version of RegionEpoch.
um... I'm a little not sure whether to consider conf-version.
Could you give me some suggestion ? @disksing @rleungx , Thank you!!

Reply:
BTW, I think there will be a problem if we use `!=` since the operator's epoch won't change after creating.